orange/src/main/java/com/hartwig/hmftools/orange/algo/purple/GermlineHeterozygousDeletionFactory.java

```diff
@@ -0,0 +1,69 @@
+package com.hartwig.hmftools.orange.algo.purple;
+
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Maps;
+import com.hartwig.hmftools.common.ensemblcache.EnsemblDataCache;
+import com.hartwig.hmftools.common.gene.TranscriptData;
+import com.hartwig.hmftools.common.purple.GeneCopyNumber;
+import com.hartwig.hmftools.common.purple.GermlineDeletion;
+import com.hartwig.hmftools.common.purple.GermlineStatus;
+import com.hartwig.hmftools.datamodel.purple.GeneProportion;
+import com.hartwig.hmftools.datamodel.purple.ImmutablePurpleHeterozygousDeletion;
+import com.hartwig.hmftools.datamodel.purple.PurpleHeterozygousDeletion;
+
+import org.jetbrains.annotations.NotNull;
+
+public class GermlineHeterozygousDeletionFactory
+{
+    @NotNull
+    private final EnsemblDataCache ensemblDataCache;
+
+    public GermlineHeterozygousDeletionFactory(@NotNull final EnsemblDataCache ensemblDataCache)
+    {
+        this.ensemblDataCache = ensemblDataCache;
+    }
+
+    @NotNull
+    public Map<PurpleHeterozygousDeletion, GermlineDeletion> mapDeletions(@NotNull List<GermlineDeletion> germlineDeletions,
+            @NotNull List<GeneCopyNumber> allSomaticGeneCopyNumbers)
+    {
+        Map<PurpleHeterozygousDeletion, GermlineDeletion> deletionMap = Maps.newHashMap();
+        for(GermlineDeletion germlineDeletion : germlineDeletions)
+        {
+            if(germlineDeletion.TumorStatus == GermlineStatus.HET_DELETION)
+            {
+                PurpleHeterozygousDeletion heterozygousDeletion = toHeterozygousDeletion(germlineDeletion, allSomaticGeneCopyNumbers);
+                if(!deletionMap.containsKey(heterozygousDeletion))
+                {
+                    deletionMap.put(heterozygousDeletion, germlineDeletion);
```

Comment:
I don't know how plausible this is biologically, but I'm not convinced that the code handles this case in the way you describe. The heterozygousDeletion variable is the key of the map, and it contains the geneProportion. I think only germline deletions of the same gene with the same gene proportion, minCopies and maxCopies would fall through the cracks, so to speak.

Reply:
You are correct indeed, and that may be sub-optional (also my initial implementation in GermlineGainLossFactory). 

My idea is to map N germline deletions to 1 PurpleGainLoss (or 1 PurpleHeterozygousDeletion) per gene, but this indeed wouldn't work if the N germline deletions have different data. 

Maybe in practice this is no problem because 2 germline deletions in a single gene would both be partial anyways, and maybe typically share other properties? Still think it would be cleaner to add a 'pickBest` function to the keySet of this map to only select the best (het) deletion for a single gene? 

Or would logic also work & make sense if 2 purple het deletions are returned for a single gene?

Reply:
I have rewritten this code to create only a single PurpleLossOfHeterozygosity object per gene.